compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -186,12 +187,8 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
     private def removeUnwantedAnnotations(sym: Symbol, metaAnnotSym: Symbol,
         metaAnnotSymBackup: Symbol, keepIfNoRelevantAnnot: Boolean)(using Context): Unit =
       def shouldKeep(annot: Annotation): Boolean =
-        val annotSym = annot.symbol
-        annotSym.hasAnnotation(metaAnnotSym)
-          || annotSym.hasAnnotation(metaAnnotSymBackup)
-          || (keepIfNoRelevantAnnot && {
-            !annotSym.annotations.exists(metaAnnot => defn.FieldAccessorMetaAnnots.contains(metaAnnot.symbol))
-          })
+        annot.hasOneOfMetaAnnotation(metaAnnotSym, metaAnnotSymBackup)
+        || keepIfNoRelevantAnnot && !annot.hasOneOfMetaAnnotation(defn.NonBeanMetaAnnots.toList*)
```

Comment:
Why exclude "bean" meta annotations here?

Reply:
This runs before bean getters and setters are created.  So if we lose the `@beanGetter`/`@beanSetter`-annotated annotations on the regular class vals, e.g.:

```scala
@(JsonProperty @beanGetter) @BeanProperty val beanGet: String = ""
```

Then they'll never be copied over onto the `def getBeanGet: String = ...` we're about to syntheses.

Reply:
In Memoize, where we split vals into fields and re-written getters, we now also drop annotations that are `beanGetter/Setter` annotated, from the field, and keep only `@getter/@setter`-annotated annotations on the (non-bean) getters/setters.

Reply:
👍 thanks